### DISCUSSION TITLE
Whats difference for .bin and .app-bin?

### DISCUSSION THREAD

**Discussion by ffps** [micropython/micropython#12613] [General]
On [download pages](https://micropython.org/download/ESP32_GENERIC_C3/)  there are files `.bin` and `.app-bin`. I could not find any description for `app-bin` files. What are they for?

**Comment by glenn20** [comment]
For ESP32 devices the `.bin` files are full firmware suitable for burning directly to the flash storage with `esptool.py`. These include a bootloader, a partition table for the flash storage and what espressif calls the "application binary". This last bit is the micropython app, which is also provided as a stand-alone firmware (the `.app-bin` file). These are useful if you want to install only the app firmware (for example if you use an OTA firmware which provides support for Over The Air firmware updates).

Also, if you compile your own micropython firmware, the `.app-bin` files are the `ports/esp32/build-*/micropython.bin` files.

**Reply by arunm-sands** [reply]
when I compile our own firmware ,two binary files are created(firmware.bin and micropython.bin) which one i should use to flash?and what is difference between  these two files?

**Reply by glenn20** [reply]
When you compile your own micropython firmware, the 'micropython.bin' is the compiled "application binary" file for micropython. This is the ".app-bin" file which is available on the micropython download page. These files are suitable for installing on a device using Over-The-Air updates.

The 'firmware.bin' file links together the espressif bootloader code, a partition table and the 'micropython.bin' file into a single firmware that can be flashed onto an esp32 device with esptool.py. This is the '.bin' file you download from the micropython download page.

So, if you are installing micropython via the USB port, you would usually use the firmware.bin and esptool.py.

If you are upgrading your firmware with OTA, you would use the micropython.bin file.

The naming conventions are a bit confusing, but do have a sort of logic to them

**Reply by arunm-sands** [reply]
thank you
